File: source/kCura.Relativity.Export/Types/ExportEventArgs.vb
Namespace kCura.Relativity.Export.Types

	Public Class ExportEventArgs
		Protected _documentsExported As Int32
		Protected _totalDocuments As Int32
		Protected _eventType As kCura.Windows.Process.EventType
		Protected _message As String
		Protected _additionalInfo As Object

#Region "Public Properties"
		Public Property DocumentsExported() As Int32
			Get
				Return _documentsExported
			End Get
			Set(ByVal value As Int32)
				_documentsExported = value
			End Set
		End Property

		Public Property TotalDocuments() As Int32
			Get
				Return _totalDocuments
			End Get
			Set(ByVal value As Int32)
				_totalDocuments = value
			End Set
		End Property

		Public Property EventType() As kCura.Windows.Process.EventType
			Get
				Return _eventType
			End Get
			Set(ByVal value As kCura.Windows.Process.EventType)
				_eventType = value
			End Set
		End Property

		Public Property Message() As String
			Get
				Return _message
			End Get
			Set(ByVal value As String)
				_message = value
			End Set
		End Property

		Public ReadOnly Property AdditionalInfo() As Object
			Get
				Return _additionalInfo
			End Get
		End Property
#End Region

		Public Sub New(ByVal documentsExported As Int32, ByVal totalDocuments As Int32, ByVal additionalInfo As Object)
			_documentsExported = documentsExported
			_totalDocuments = totalDocuments
			_additionalInfo = additionalInfo
		End Sub

		Public Sub New(ByVal documentsExported As Int32, ByVal totalDocuments As Int32, ByVal message As String, ByVal eventType As kCura.Windows.Process.EventType, ByVal additionalInfo As Object)
			_documentsExported = documentsExported
			_totalDocuments = totalDocuments
			_message = message
			_eventType = eventType
			_additionalInfo = additionalInfo
		End Sub
	End Class
End Namespace

File: source/kCura.Relativity.Export.ConvertedToC#/Types/ExportEventArgs.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Data;
using System.Linq;
namespace kCura.Relativity.Export.Types
{

	public class ExportEventArgs
	{
		protected Int32 _documentsExported;
		protected Int32 _totalDocuments;
		protected kCura.Windows.Process.EventType _eventType;
		protected string _message;

		protected object _additionalInfo;
		#region "Public Properties"
		public Int32 DocumentsExported {
			get { return _documentsExported; }
			set { _documentsExported = value; }
		}

		public Int32 TotalDocuments {
			get { return _totalDocuments; }
			set { _totalDocuments = value; }
		}

		public kCura.Windows.Process.EventType EventType {
			get { return _eventType; }
			set { _eventType = value; }
		}

		public string Message {
			get { return _message; }
			set { _message = value; }
		}

		public object AdditionalInfo {
			get { return _additionalInfo; }
		}
		#endregion

		public ExportEventArgs(Int32 documentsExported, Int32 totalDocuments, object additionalInfo)
		{
			_documentsExported = documentsExported;
			_totalDocuments = totalDocuments;
			_additionalInfo = additionalInfo;
		}

		public ExportEventArgs(Int32 documentsExported, Int32 totalDocuments, string message, kCura.Windows.Process.EventType eventType, object additionalInfo)
		{
			_documentsExported = documentsExported;
			_totalDocuments = totalDocuments;
			_message = message;
			_eventType = eventType;
			_additionalInfo = additionalInfo;
		}
	}
}
